app/controllers/api/v1/targets_controller.rb

```diff
@@ -0,0 +1,35 @@
+module Api
+  module V1
+    class TargetsController < Api::V1::ApiController
+      before_action :fetch_user, only: [:create]
+      before_action :fetch_topic, only: [:create]
+
+      def create
+        target_params = create_params.except(:user, :topic)
+        @target = Target.new(target_params)
+        @target.user = @user
+        @target.topic = @topic
+        render_error_response unless @target.save
+      end
+
+      private
+
+      def fetch_user
+        @user = User.find(create_params[:user][:id].to_i)
+      end
+
+      def fetch_topic
+        @topic = Topic.find(create_params[:topic][:id].to_i)
+      end
+
+      def create_params
+        params.require(:target).permit(:title, :radius, :latitude, :longitude, user: [:id],
+                                                                               topic: [:id])
```

Comment:
```suggestion
        params.require(:target).permit(:title, :radius, :latitude, :longitude, user: :id,
                                                                               topic: :id)
```
Just because there is only 1 element in the array